network/p2p/dns/resolver.go

```diff
@@ -0,0 +1,184 @@
+package dns
+
+import (
+	"context"
+	"net"
+	"sync"
+	"time"
+	_ "unsafe" // for linking runtimeNano
+
+	madns "github.com/multiformats/go-multiaddr-dns"
+
+	"github.com/onflow/flow-go/module"
+)
+
+//go:linkname runtimeNano runtime.nanotime
+func runtimeNano() int64
```

Comment:
Using `runtime.runtimeNano()` instead of `time.Now()` per [recommendation](https://www.notion.so/dapperlabs/Time-Stamp-Counter-vs-time-Now-a511a7756a6b4259a7229aedbf9c6f93#d1674459e7f94c8faca346e0d1dc95d7
11:54
) of @simonhf